### DISCUSSION TITLE
The Long-Term Plan of gPRC Exhook

### DISCUSSION THREAD

**Discussion by Nextdrive-DomingoTsao** [emqx/emqx#12859] [Q&A] [answered]
Dear EMQX Team,

I hope this message finds you well. We are currently utilizing EMQX version **EMQX v5.0.x** in our production env and are exploring the development of custom EMQX lifecycle hooks leveraging `gRPC Exhook` for a new feature on authentication.

Upon reviewing a [previous discussion](https://github.com/emqx/emqx/discussions/10121#discussioncomment-5295411), it came to our attention that there might be plans to **deprecate** `gRPC Exhook` in the future. This information has led us to seek further clarification on a few critical points:

- Is deprecation of `gRPC Exhook` currently planned? If so, could you provide any anticipated timelines or specific versions this might affect?
- Have there been any reported issues or challenges from developers who are using `gRPC Exhook` that we should consider besides #5218?
- In light of the potential deprecation, do you recommend proceeding with the development of custom hooks using `gRPC Exhook` for a new feature?
Your insights into these questions would be invaluable to us as we plan our development strategy. We appreciate any guidance or recommendations you can provide.

Thank you very much for your support and assistance.

Best regards,

Doming Tsao

**Comment by zmstone** [comment, marked as answer]
Apologies for the confusion. I meant to say exproto but not exhook.
There is no plan to deprecate exhook.

**Reply by zmstone** [reply]
also clarified here: https://github.com/emqx/emqx/discussions/10121#discussioncomment-9083478

**Reply by Nextdrive-DomingoTsao** [reply]
Glad to hear it! Thank you once again for your prompt and invaluable assistance. I also want to express my gratitude for your ongoing contributions to the community !
